model/inline/src/main/scala/aqua/model/inline/MakeRawInliner.scala

```diff
@@ -0,0 +1,63 @@
+package aqua.model.inline
+
+import aqua.model.{
+  CallModel,
+  CallServiceModel,
+  LiteralModel,
+  OpModel,
+  SeqModel,
+  ValueModel,
+  VarModel
+}
+import aqua.model.inline.raw.RawInliner
+import cats.data.Chain
+import aqua.model.inline.state.{Arrows, Exports, Mangler}
+import aqua.raw.value.{MakeStructRaw, LiteralRaw}
+import cats.data.{NonEmptyMap, State}
+import aqua.model.inline.Inline
+import aqua.model.inline.RawValueInliner.{unfold, valueToModel}
+import aqua.types.ScalarType
+import cats.syntax.traverse.*
+import cats.syntax.monoid.*
+import cats.syntax.functor.*
+import cats.syntax.flatMap.*
+import cats.syntax.apply.*
+
+object MakeRawInliner extends RawInliner[MakeStructRaw] {
+
+  private def createObj(fields: NonEmptyMap[String, ValueModel], result: VarModel): OpModel.Tree = {
+    val args = fields.toSortedMap.toList.flatMap { case (name, value) =>
+      LiteralModel.fromRaw(LiteralRaw.quote(name)) :: value :: Nil
+    }
+    CallServiceModel(
+      LiteralModel("\"json\"", ScalarType.string),
+      "obj",
+      CallModel(
+        args,
+        CallModel.Export(result.name, result.`type`) :: Nil
+      )
+    ).leaf
+  }
+
+  override def apply[S: Mangler: Exports: Arrows](
+    raw: MakeStructRaw,
+    propertiesAllowed: Boolean
+  ): State[S, (ValueModel, Inline)] = {
+    for {
+      name <- Mangler[S].findAndForbidName(raw.name + "_obj")
```

Comment:
with `use "smth.aqua"` you can get to situation when this `raw.name` contains dots, like `Smth.Struct`.